test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -806,22 +807,23 @@
       (is (utils/public-key? key))
       (is (= 768 (utils/keylength key)))))))
 
-(deftest initialize-master-ssl!-test-with-incorrect-keylength
-  (let [tmp-confdir (fs/copy-dir confdir (ks/temp-dir))
-        settings (testutils/master-settings tmp-confdir)
-        ca-settings (testutils/ca-settings (str tmp-confdir "/ssl/ca"))]
+(when-not (SSLUtils/isFIPS)
+  (deftest initialize-master-ssl!-test-with-incorrect-keylength
+    (let [tmp-confdir (fs/copy-dir confdir (ks/temp-dir))
+          settings (testutils/master-settings tmp-confdir)
+          ca-settings (testutils/ca-settings (str tmp-confdir "/ssl/ca"))]
 
-    (retrieve-ca-cert! (:cacert ca-settings) (:localcacert settings))
+      (retrieve-ca-cert! (:cacert ca-settings) (:localcacert settings))
 
-    (testing "should throw an error message with too short keylength"
-      (is (thrown?
-           InvalidParameterException
-           (initialize-master-ssl! (assoc settings :keylength 128) "master" ca-settings))))
```

Comment:
we probably still want a guard in `initialize-master-ssl!` to not allow keys < 1024 bits in FIPS mode, because if it ever happens, the process will effectively be dead, but it won't actually exit because it'll trip the error handler in bc-fips.

Reply:
I pulled that out thinking that to do it well we actually should add the keylength verification to a different part of our code (and in our puppetsever-ca cli tool which is recommended over the `initialize-master-ssl` method), that those changes aren't something we should necessarily do in a z, and while the UX is bad it is pretty unlikely (someone would have to install FIPS and then manually configure the keylength to be less than 1024?

My thinking was that we could clean that up in the next 'Y' release: https://tickets.puppetlabs.com/browse/SERVER-2655

Reply:
maybe we do that in jvm-ssl-utils, where the calls into the bc provider happen?